src/navigation/__tests__/Tabs.test.tsx

```diff
@@ -0,0 +1,30 @@
+import { render, cleanup } from "@testing-library/react"
+import Tabs from "../Tabs"
+
+afterEach(cleanup)
+
+describe("<Tabs>", () => {
+  it("outputs the right markup", () => {
+    const { container, getByText } = render(
+      <Tabs defaultIndex={1}>
+        <Tabs.TabList>
+          <Tabs.Tab className="other-tab">Other</Tabs.Tab>
+          <Tabs.Tab className="default-tab">Default</Tabs.Tab>
+        </Tabs.TabList>
+        <Tabs.TabPanel className="other-panel">OtherPanel</Tabs.TabPanel>
+        <Tabs.TabPanel className="default-panel">DefaultPanel</Tabs.TabPanel>
+      </Tabs>
+    )
+
+    expect(getByText("Other")).toBeTruthy()
+    expect(container.getElementsByClassName("other-tab").length).toBe(1)
+    expect(container.getElementsByClassName("other-panel").length).toBe(1)
+
+    expect(getByText("Default")).toBeTruthy()
+    expect(container.getElementsByClassName("default-tab").length).toBe(1)
+    expect(container.getElementsByClassName("default-panel").length).toBe(1)
+
+    expect(container.querySelectorAll(".tabs-tab.is-active").length).toBe(1)
+    expect(container.querySelectorAll(".tabs-panel.is-active").length).toBe(1)
+  })
```

Comment:
You should add an additional expect statements to check that "OtherPanel" is not in the document and "DefaultPanel" is in the document. And we could also get extra coverage by testing the click functionality on the "Other tab"

Reply:
Good idea

Reply:
@ludtkemorgan actually, so to clarify the content of unselected tabs is still present in the document, it's just hidden via CSS. This is what we want, because if we have a form spread across multiple tabs, we want the entire form to be available for validation & serialization. But I can go ahead and add the click event testing and make sure only one tab at a time is being shown.

Reply:
That makes sense. There is a chance that having everything in the document could cause some performance issues if there are a lot of items in the tabs. Especially if an api call is needed to display something just in the hidden tab. I haven't dug into the why, but for example the listing add/edit page is fairly slow to load and theoretically we only need to display one tab at a time. I'm ok with this for now, however in the past this sort of thing has caused issues on other projects I've worked on.